sdk/dapp-kit/src/hooks/useIotaClientQuery.ts

```diff
@@ -72,3 +105,40 @@ export function useIotaClientQuery<
         },
     });
 }
+
+export function useIotaClientSuspenseQuery<
+    T extends keyof IotaRpcMethods,
+    TData = IotaRpcMethods[T]['result'],
+>(
+    ...args: undefined extends IotaRpcMethods[T]['params']
+        ? [
+              method: T,
+              params?: IotaRpcMethods[T]['params'],
+              options?: UndefinedInitialDataOptions<TData>,
+          ]
+        : [
+              method: T,
+              params: IotaRpcMethods[T]['params'],
+              options?: UndefinedInitialDataOptions<TData>,
+          ]
+) {
+    const [method, params, options = {}] = args as [
+        method: T,
+        params?: IotaRpcMethods[T]['params'],
+        options?: UndefinedInitialDataOptions<TData>,
+    ];
+
+    const iotaContext = useIotaClientContext();
+
+    const query = useMemo(() => {
+        return getIotaClientQuery<T>({
+            client: iotaContext.client,
+            network: iotaContext.network,
+            method,
+            params,
+            options,
+        });
+    }, [iotaContext.client, iotaContext.network, method, params, options]);
```

Comment:
```suggestion
    }, [iotaContext.network, method, params, options]);
```
> we dont include anywhere else on purpose because its an expensive object to memoize

> I find it weird that it was included in there

> You can see that we dont put it in neither getIotaClientQuery or useIotaClientQuery  or any other custom hook across all our packages and apps